linz2osm/convert/osm.py

```diff
@@ -38,15 +38,33 @@ def get_srtext_from_srid(srid):
 
 def get_layer_geometry_type(database_id, layer):
     """ Returns a (geometry_type, srid) tuple """
-    cursor = connections[database_id].cursor()
-    cursor.execute('SELECT type,srid FROM geometry_columns WHERE f_table_name=%s;', [layer.name])
-    r = cursor.fetchone()
-    return tuple(r[:2])
+    if layer.geometry_type == 'RELATION':
+        geotypes = get_relation_geometry_types(database_id, layer)
+        if not geotypes:
+            raise ValueError("No members for relation")
+        srid = geotypes.values()[0][1]
+        if not all([srid == s for (geotype, s) in geotypes.values()]):
+            raise ValueError("SRIDs differ for relation layers!")
+        return ("GEOMETRYCOLLECTION", srid)
```

Comment:
* We're checking that they're all the same SRID, not that they're all the same geotype. Indeed, our one use case has a geometry collection of one point and two linestrings.

* The relation layer geotype is "RELATION", rather than a regular geotype. The actual geometry is always returned as a GeometryCollection. There's no separate record of the actual geotype, other than going through the Member relation and getting the geotypes of the members.